app/controllers/external_works_controller.rb

```diff
@@ -21,7 +21,8 @@ def fetch
 
   def index
     if params[:show] == 'duplicates'
-      @external_works = ExternalWork.duplicate.order("created_at DESC").paginate(page: params[:page])
+      works = ExternalWork.all.group(:url).having("count(*) >1").order("created_at DESC")
```

Comment:
Since the root issue seems to be that the `duplicate` scope for `ExternalWork`s doesn't separate out `group` and `having` the way you did here, why not fix the scope instead of changing this code?

Reply:
I don't know how to do that, @tickinginstant . I have looked at this and got something similar. However using count/size fails as the count(*) bit makes the sql invalid, unlike the earlier statement. I have worked out that if you use length the whole set is loaded in to memory which can then be counted. Which does work. I suspect size mostly worked as the whole query had been loaded in to memory before making size the same as length.

Reply:
What I meant by fixing the scope is editing the `ExternalWork` scope to be:
```
scope :duplicate, -> { group(:url).having("count(DISTINCT id) > 1") }
```
For me, that fixes the issue with the SQL error when calling `ExternalWork.duplicate.count`.

Because it's a `group`/`count` combo, it returns a hash instead of a number, which makes the "Show duplicates" button display wrong. But it should work fine with the `will_paginate` gem, which [calls `size` on the hash returned by `count` in order to get the true size](https://github.com/mislav/will_paginate/blob/b6fb7b0db9cacb858ab7092a0447ff9db573b9fd/lib/will_paginate/active_record.rb#L68-L79). Retrieving tens or hundreds of thousands of `ExternalWork` URLs isn't super-efficient, but it's better than calling `length`, which will retrieve more columns from the database, and take more time to stuff the data into `ActiveRecord` objects.

Reply:
I had tried that, but I have tried again and I get a test failure: https://travis-ci.org/zz9pzza/otwarchive/jobs/556960242 on this branch: https://github.com/zz9pzza/otwarchive/tree/AO3-4742_2

Reply:
I get  Show duplicates ({\"http://zooey-glass.dreamwidth.org\"=>3}) which mirrors what I get in the console.

Reply:
Yes, as I said:

>Because it's a group/count combo, it returns a hash instead of a number, which makes the "Show duplicates" button display wrong.

But it's easy enough to fix the same way that the `will_paginate` gem handles it: by calling `count.size` instead of just `count` to figure out the true size. (And as I said in my previous comment, that's more efficient than calling `length` in the `external_works/index` view.)